lib/ofp-print.c

```diff
@@ -3327,6 +3327,94 @@ ofp_print_nxst_ipfix_flow_reply(struct ds *string, const struct ofp_header *oh)
     }
 }
 
+static void
+ofp_print_hex(struct ds *string, void *arg, unsigned int n, const char *sep)
+{
+    unsigned char *data = arg;
+    unsigned int i;
+
+    for (i = 0; i < n; i++) {
+        const char *pfx = "";
+
+        if (!i)
+            /* nothing */;
+        else if (!(i % 16))
+            ds_put_cstr(string, "\n");
+        else if (!(i % 8))
+            ds_put_cstr(string, "  ");
+        else
+            pfx = sep;
+
+        ds_put_format(string, "%s%02x", i ? pfx : "", data[i]);
+    }
+}
+
+static void
+ofp_print_decimal(struct ds *string, void *arg, unsigned int n, const char *sep)
+{
+    unsigned char *data = arg;
+    unsigned int i;
+
+    for (i = 0; i < n; i++) {
+        const char *pfx = "";
+
+        if (!i)
+            /* nothing */;
+        else if (!(i % 16))
+            ds_put_cstr(string, "\n");
+        else if (!(i % 8))
+            ds_put_cstr(string, "  ");
+        else
+            pfx = sep;
+
+        ds_put_format(string, "%s%u", i ? pfx : "", data[i]);
+    }
+}
+
+static void
+ofp_print_dump_map_element(struct ds *string, struct ol_dump_map_reply *buffer, void *data, int verbosity)
+{
+    ds_put_cstr(string, "Key: \n");
+    if (verbosity < 2) {
```

Comment:
I used a method `ofp_print` which takes as an argument `int verbosity`. What I see is that every reply is printed via ofp-print. My intention was not to break this principle. But I agree that it is not a beautiful piece of code. Do you have any ideas?

Reply:
`ofp_print` actually uses `verbosity` to indicate verbosity: there are several levels and the details of the printed packet depend on the passed level. Here, as far as I can see, it has a binary value: 0 or 3. I think it could be replaced with a simple `bool hex` or `bool print_hex`.